app/forms/declaration/additionaldocuments/DocumentWriteOff.scala

```diff
@@ -16,44 +16,79 @@
 
 package forms.declaration.additionaldocuments
 
-import play.api.data.Forms.{bigDecimal, optional, text}
-import play.api.data.{Form, Forms}
+import play.api.data.Forms.{optional, text}
+import play.api.data.{Form, FormError, Forms}
 import play.api.libs.json.Json
-import utils.validators.forms.FieldValidator.{isAlphanumericWithSpecialCharacters, lengthInRange}
+import utils.validators.forms.FieldValidator._
 
 case class DocumentWriteOff(measurementUnit: Option[String], documentQuantity: Option[BigDecimal])
 
 object DocumentWriteOff {
 
+  def convert(measurementUnit: Option[String], documentQuantity: Option[String]): DocumentWriteOff =
+    new DocumentWriteOff(measurementUnit, documentQuantity.map(BigDecimal(_)))
+
   implicit val format = Json.format[DocumentWriteOff]
 
-  private val measurementUnitMinLength = 3
-  private val measurementUnitMaxLength = 5
+  private val measurementUnitLength = 3
+  private val qualifierLength = 1
   private val documentQuantityMaxLength = 16
   private val documentQuantityMaxDecimalPlaces = 6
 
+  val documentWriteOffKey = "documentWriteOff"
   val measurementUnitKey = "measurementUnit"
+  val qualifierKey = "qualifier"
   val documentQuantityKey = "documentQuantity"
 
+  private def form2Model: (Option[String], Option[String], Option[String]) => DocumentWriteOff = {
+    case (None, None, quantity)      => convert(None, quantity)
+    case (unit, None, quantity)      => convert(unit, quantity)
+    case (unit, qualifier, quantity) => convert(Some(Seq(unit, qualifier).flatten.mkString("#")), quantity)
+  }
+
+  private def model2Form: DocumentWriteOff => Option[(Option[String], Option[String], Option[String])] =
+    model => {
+      val unitAndQualifier: Option[Array[String]] = model.measurementUnit.map(_.split("#"))
+      Some(unitAndQualifier.flatMap(_.headOption), unitAndQualifier.flatMap(_.lift(1)), model.documentQuantity.map(_.toString()))
```

Comment:
What `lift` is doing? Can't find it in documentation.

Reply:
It's getting the second (index 1) optional String from the array.   I probably copies that from somewhere.  I also had lift(0) but InteliJ told me that if can be replaced with headOption.

Specifically - https://stackoverflow.com/questions/17965059/what-is-lifting-in-scala